### DISCUSSION TITLE
Seeing iwasm output when exiting Python's REPL

### DISCUSSION THREAD

**Discussion by brettcannon** [bytecodealliance/wasm-micro-runtime#1886] [Q&A]
I was wondering if someone knew why iwasm prints those 14 lines when Python exits using its `exit()` function? It doesn't happen when using e.g. `-c pass` or various other ways for Python to exit. Is the output some debugging information about something?

```
scratch/wasi/Python-3.11.0-wasm32-wasi-16 via C v9.4.0-gcc
❯ iwasm --version
iwasm 1.1.2

scratch/wasi/Python-3.11.0-wasm32-wasi-16 via C v9.4.0-gcc
❯ iwasm --dir=. python.wasm -c "exit()"

#00 proc_exit
#01 $f7893
#02 $f7770
#03 $f7913
#04 $f3646
#05 $f3710
#06 $f3719
#07 $f4018
#08 $f4021
#09 $f4023
#10 $f45
#11 $f7849
#12 $f44
#13 _start


scratch/wasi/Python-3.11.0-wasm32-wasi-16 via C v9.4.0-gcc
❯ echo $status
0
```

**Comment by tonibofarull** [comment]
How have you generated the `python.wasm`? Please, provide information to reproduce the behavior!

**Reply by brettcannon** [reply]
You can download it from https://github.com/tiran/cpython-wasm-test/releases/tag/v3.11.0 or you can do a build from scratch following the instructions at https://github.com/python/cpython/tree/main/Tools/wasm .

**Comment by yamt** [comment]
> Is the output some debugging information about something?

your build has `WAMR_BUILD_DUMP_CALL_STACK` enabled and it's the output from it.

**Reply by brettcannon** [reply]
Thanks for the info! The build came from Homebrew, and they do indeed [turn that flag on](https://github.com/Homebrew/homebrew-core/blob/fe7ce49dc9076d76465e2a379d35423de859237d/Formula/wasm-micro-runtime.rb#L37).
